tests/tests.py

```diff
@@ -141,17 +141,27 @@ def test_insert_field_add_import():
 
 def is_as_expected(path_to_scenario, output_summary):
     expected_output = join(path_to_scenario, "expected")
+    input_dir = join(path_to_scenario, "input")
     for file_name in listdir(expected_output):
         with open(join(expected_output, file_name), "r") as f:
             file_content = f.read()
             expected_content = "".join(file_content.split())
+
+            # Search for the file in the output summary
             updated_content = [
                 "".join(o.content.split())
                 for o in output_summary
                 if basename(o.path) == file_name
-            ][0]
-            if expected_content != updated_content:
-                return False
+            ]
+            # Check if the file was rewritten
+            if updated_content:
+                if expected_content != updated_content[0]:
+                    return False
+            else:
+                # The scenario where the file is not expected to be rewritten
+                original_content= Path(join(input_dir, file_name)).read_text()
+                if expected_content != "".join(original_content.split()):
+                    return False
```

Comment:
So, technically this check is just ensuring that we propagate files from `input/` to `expected/` when Piranha is not supposed to rewrite them, right? It allows us to have files in our tests that Piranha is not supposed to modify.

The actual new test cases are the added files above, which we just want to make sure don't get extraneous imports added to (because the rule will never run on them, since it's not a seed rule, nor triggered from another rule running on those same files).

Am I getting all of this right?

Reply:
Ok, this all makes sense and is what I thought (But nicely explained! Thanks!).

No further review needed from me on this one unless there are new changes, btw. LGTM and I already approved.